components/raftstore/src/store/async_io/write_router.rs

```diff
@@ -158,13 +163,21 @@ where
         if self.last_unpersisted.is_some() {
             return false;
         }
-        if ctx.config().store_io_pool_size <= 1 {
+        let async_io_pool_size = ctx.write_senders().capacity();
+        if async_io_pool_size <= 1 {
             self.writer_id = 0;
             return true;
         }
-        if last_unpersisted.is_none() {
+        let need_correct_writer_id = {
+            // Because the given `writer_id` is not atomically updated during the
+            // progress of resizing, that is, the given `writer_id` is generated by
+            // `% senders.capacity()` and not consistent with current capacity of
+            // senders. It should be corrected when resizing the size of senders.
+            self.writer_id >= async_io_pool_size
+        };
+        if last_unpersisted.is_none() || need_correct_writer_id {
```

Comment:
Need to make sure changing the write_id when last_unpersisted is not none does not have correctness issue.
Also if that's safe, then the comment below needs to be updated. 
`// If no previous pending ready, we can randomly select a new writer worker`

Reply:
Currently, the corner case:
> When decreasing threads, there might still have some not persisted tasks in writers, and this will cause latent write inconsistency. 
has been worked through by "do not releasing the decreased async-ios" to make sure the previous unpersisted tasks, which have been already assigned a `writer_id` with old `async_io_pool_size`, could be persisted correctly.